### PR TITLE
Proper test for poisson random time

### PR DESCRIPTION
**Pull request by mschauer**
@SebaGraz Should help with the task tomorrow.

### CODE REVIEW

File: test/poisson.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+function F((a,b,c), π)
+    if a <= 0 && (a + π*b <= 0)
+        return π*c
+    elseif a > 0 && (a + π*b < 0)
+        return π*c - a^2/(2*b)
+    elseif a > 0 && (a + π*b >= 0)
+        return 1/2*π*(2*a + π*b + 2*c) 
+    else # a <= 0 && (a + π*b > 0)
+        return a^2/(2*b) + π*a + (π^2*b)/2 + π*c 
+        #return (π*b*(π*b + 2*c) - 2*(-a)*π*b + a^2)/(2b)
```
Comment (SebaGraz) [inline test/poisson.jl:16]:
```suggestion
```

File: src/poissontime.jl [hunk 2]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+
+Obtaining waiting time for inhomogeneous Poisson Process
+with rate of the form λ(t) = c + (a + b*t)^+, 
+where `c`> 0 ,`a, b` ∈ R, `u` uniform random variable
+"""
+function poisson_time((a,b,c)::NTuple{3}, u)
+    if b > 0
+        if a < 0
+            if !(-c*a/b + log(u) >= 0)
+                # return sqrt((-log(u) + a^2/(2*b))*2.0/b) - (a+c)/b
```
Comment (SebaGraz) [inline src/poissontime.jl:43]:
```suggestion
```

File: src/poissontime.jl [hunk 3]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+                return sqrt(-2*b*log(u) + c^2 + 2*a*c)/b - (a+c)/b
+            else
+                return -log(u)/c
+            end
+        else # a[i]>0 OK
```
Comment (SebaGraz) [inline src/poissontime.jl:48]:
```suggestion
        else # a >0
```

File: src/poissontime.jl [hunk 4]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+            return sqrt(-log(u)*2.0*b + (a+c)^2)/b - (a+c)/b
+        end
+    elseif b == 0
+        if a > 0
+            return -log(u)/(a + c)
+        else # a[i] <= 0
+            return -log(u)/c
+        end
+    else # b[i] < 0
+        if a <= 0 # OK
```
Comment (SebaGraz) [inline src/poissontime.jl:58]:
```suggestion
        if a <= 0.0 
```

File: src/poissontime.jl [hunk 5]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+            else
+                return -log(u)/c
+            end
+        else # a[i]>0 OK
+            return sqrt(-log(u)*2.0*b + (a+c)^2)/b - (a+c)/b
+        end
+    elseif b == 0
+        if a > 0
+            return -log(u)/(a + c)
+        else # a[i] <= 0
```
Comment (SebaGraz) [inline src/poissontime.jl:54]:
```suggestion
        else # a <= 0
```

File: src/poissontime.jl [hunk 6]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+            return -log(u)/c
+        end
+    else # b[i] < 0
```
Comment (SebaGraz) [inline src/poissontime.jl:57]:
```suggestion
    else # b < 0
```

File: src/poissontime.jl [hunk 7]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+        if a <= 0 # OK
+            return -log(u)/c
+        elseif  !(- c*a/b - a^2/(2*b)  + log(u) <= 0.0)
```
Comment (SebaGraz) [inline src/poissontime.jl:60]:
```suggestion
        elseif  - c*a/b - a^2/(2*b)  + log(u) > 0.0
```

File: src/poissontime.jl [hunk 8]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+    poisson_time((a, b, c), u)
+
+Obtaining waiting time for inhomogeneous Poisson Process
+with rate of the form λ(t) = c + (a + b*t)^+, 
+where `c`> 0 ,`a, b` ∈ R, `u` uniform random variable
+"""
+function poisson_time((a,b,c)::NTuple{3}, u)
+    if b > 0
+        if a < 0
+            if !(-c*a/b + log(u) >= 0)
```
Comment (SebaGraz) [inline src/poissontime.jl:42]:
```suggestion
            if -c*a/b + log(u) <= 0.0
```

File: src/poissontime.jl [hunk 9]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+                # return sqrt((-log(u) + a^2/(2*b))*2.0/b) - (a+c)/b
+                return sqrt(-2*b*log(u) + c^2 + 2*a*c)/b - (a+c)/b
+            else
+                return -log(u)/c
+            end
+        else # a[i]>0 OK
+            return sqrt(-log(u)*2.0*b + (a+c)^2)/b - (a+c)/b
+        end
+    elseif b == 0
+        if a > 0
```
Comment (SebaGraz) [inline src/poissontime.jl:52]:
```suggestion
        if a > 0.0
```

File: src/poissontime.jl [hunk 10]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
 
+"""
+    poisson_time((a, b, c), u)
+
+Obtaining waiting time for inhomogeneous Poisson Process
+with rate of the form λ(t) = c + (a + b*t)^+, 
+where `c`> 0 ,`a, b` ∈ R, `u` uniform random variable
+"""
+function poisson_time((a,b,c)::NTuple{3}, u)
+    if b > 0
```
Comment (SebaGraz) [inline src/poissontime.jl:40]:
```suggestion
    if b > 0.0
```

File: src/poissontime.jl [hunk 11]
```diff
@@ -29,6 +29,42 @@ function poisson_time(a, b, u)
...
+        if a < 0
```
Comment (SebaGraz) [inline src/poissontime.jl:41]:
```suggestion
        if a < 0.0
```

**Review by SebaGraz — COMMENTED** [review]
clean up.

File: src/poissontime.jl (hunk 8 above)
Comment (mschauer) [reply src/poissontime.jl:42]: `<` or `<=`?

Comment (SebaGraz) [reply src/poissontime.jl:42]: Maybe better `<` (but it should be equivalent)

Comment (mschauer) [reply src/poissontime.jl:42]: You can edit your comment to change it

Comment (mschauer) [reply src/poissontime.jl:42]: I could

**Pull request merged**